### PR TITLE
Add text-to-speech tutorial

### PR DESCRIPTION
**Pull request by mthrok**
Supersedes #1664

### CODE REVIEW

**Review by nateanl — APPROVED** [review]
LGTM. Same as wav2vec2, adding colab link should be better?

File: intermediate_source/text_to_speech_with_torchaudio_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+# Since the pre-trained Tacotron2 model expects specific set of symbol
+# tables, the same functionalities available in ``torchaudio``. This
+# section is more for the explanation of the basis of encoding.
+# 
+# Firstly, we define the set of symbols. For example, we can use
+# ``'_-!\'(),.:;? abcdefghijklmnopqrstuvwxyz'``. Then, we will map the
+# each character of the input text into the index of the corresponding
+# symbol in the table.
+# 
+# The following is an exmaple of such processing. In the example, symbols
```
Comment (carolineechen) [inline intermediate_source/text_to_speech_with_torchaudio_tutorial.py:94]:
```suggestion
# The following is an example of such processing. In the example, symbols
```

File: intermediate_source/text_to_speech_with_torchaudio_tutorial.py [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+def text_to_sequence(text):
+  text = text.lower()
+  return [look_up[s] for s in text if s in symbols]
+
+text = "Hello world! Text to speech!"
+print(text_to_sequence(text))
+
+
+######################################################################
+# As mentioned in the above, since the symbol table and indices must match
```
Comment (carolineechen) [inline intermediate_source/text_to_speech_with_torchaudio_tutorial.py:111]:
```suggestion
# As mentioned in the above, the symbol table and indices must match
```

File: intermediate_source/text_to_speech_with_torchaudio_tutorial.py [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+# what the pretrained Tacotron2 model expects. ``torchaudio`` provides the
+# such transform along with the pretrained model. For example, you can
```
Comment (carolineechen) [inline intermediate_source/text_to_speech_with_torchaudio_tutorial.py:113]:
```suggestion
# transform along with the pretrained model. For example, you can
```

**Review by carolineechen — APPROVED** [review]

**Pull request merged**